src/core/gobjectptr.h

```diff
@@ -0,0 +1,74 @@
+#ifndef FM2_GOBJECTPTR_H
+#define FM2_GOBJECTPTR_H
+
+#include <glib.h>
+#include <glib-object.h>
+#include <cstddef>
+
+namespace Fm2 {
+
+template <typename T>
+class GObjectPtr {
```

Comment:
Didn't you consider simplification of the `GObjectPtr` by using smart pointers? E.g.:
```c++
template <typename T>
class GObjectPtr : private std::shared_ptr<T> {
public:
    GObjectPtr(T* gobj = nullptr, bool add_ref = true) : std::shared_ptr<T>{gobj, [this] { if (*this) g_object_unref(get()); } } {
        if (add_ref && *this)
            g_object_ref(get());
    }
    GObjectPtr(const GObjectPtr& r) : std::shared_ptr<T>(r) {};
    GObjectPtr(const GObjectPtr&& r) : std::shared_ptr<T>(r) {};
    using std::shared_ptr<T>::get;
    using std::shared_ptr<T>::operator =;
    using std::shared_ptr<T>::operator ==;
    using std::shared_ptr<T>::operator !=;
    using std::shared_ptr<T>::operator bool;
};
```

Yes, this way `GObjectPtr` family increases the g_object_ref only once (the first time the shared_ptr is constructed), but the shared_ptr implementation assures that the g_object_unref is called also only when the last one is destructed.